### DISCUSSION TITLE
Invisible GitHub Apps with write access - How to obtain information?

### DISCUSSION THREAD

**Discussion by hwine** [community/community#24815] [Apps, API and Webhooks]
<p>As an org owner using the <a href="https://developer.github.com/changes/2019-10-23-get-installations-for-org/" rel="nofollow noopener">get installations (beta)</a> v3 endpoint, I receive information about installations that are not visible in the web UI. Furthermore, the “html_url” returns a 404. (With a visible GitHub App, this url links to the installation information page for that org.) These invisilbe apps have the permissions to write repository content!</p>
<p>I’ve noticed that “29110” is the value for the “app_id” of almost all of these invisible apps. I see this invisible app on almost all of the organizations I have owner access to. My guess is that app 29110 is somehow involved in the operation of GitHub Actions, but I’d love to:</p>
<ul>
<li>get confirmation from someone at GitHub</li>
<li>find out how to detect “internal GitHub” apps programatically.</li>
</ul>
<p>Invisible apps with write permissions to source code make me nervous. Has anyone else run into this?</p>

**Comment by seveas** [comment, marked as answer]
<p>Hi <a class="mention" href="/hwine">@hwine</a>,</p>
<p>I can confirm that 29110 is a GitHub app owned by github itself. I don’t know why the API doesn’t return the right information here, that would be worth asking support.</p>

**Comment by hwine** [comment]
<p>Thanks! I’ll sleep better this weekend. :smiley:</p>
<p>How/where did you determine it was a GitHub owned app? (I’d like to learn how to fish.)</p>
<p>Enjoy your weekend!</p>
<p>–Hal</p>
<p>P.S. I do have a support ticket in, but don’t really expect a reply until next week. Post Universe recovery for much of the staff I assume.</p>

**Comment by seveas** [comment]
<p>That fish unfortunately needs to be caught with a very special net, for which one needs to be employed by GitHub. I rarely use that net when helping people on the forum, but this sounded potentially dangerous so I looked in some internal systems to see what this was.</p>

**Comment by hwine** [comment]
<p>Update: The API endpoint has been fixed to no longer return these “internal apps”. \o/</p>
